src/app/store/assets/assets.utils.ts

```diff
@@ -0,0 +1,59 @@
+import BigNumber from 'bignumber.js';
+
+import { StacksAssetFt } from '@shared/models/asset.model';
+import { formatContractId } from '@app/common/utils';
+
+export function mergeAssetBalances(
+  anchoredAssets: StacksAssetFt[],
+  unanchoredAssets: StacksAssetFt[],
+  assetType: string
+) {
+  const assetMap = new Map();
+  // Merge both balances (unanchored and anchored)
+  anchoredAssets.forEach(
+    asset =>
+      asset.type === assetType &&
+      assetMap.set(formatContractId(asset.contractAddress, asset.contractName), {
+        ...asset,
+        ...{ subBalance: new BigNumber(0) },
+      })
+  );
+  unanchoredAssets.forEach(asset => {
+    if (asset.type !== assetType) return;
+    const key = formatContractId(asset.contractAddress, asset.contractName);
+    const match = assetMap.get(key);
+    if (match) {
+      match.subBalance = asset?.balance;
+    } else {
+      assetMap.set(key, {
+        ...asset,
+        subBalance: asset.balance,
+        balance: new BigNumber(0),
+      });
+    }
+  });
+  return [...assetMap.values()] as StacksAssetFt[];
```

Comment:
As we're returning array, rather than an obj/dictionary/map, I would avoid use of map like this (sorry if it wasn't clear in last comment, I meant `Array.map`, not `new Map`). 

For general consistency, I'm pushing that we try following a declarative style, immutable array methods. For example, here you _could_ imperatively do:

```ts
let arr = [];
for (let = 0; i < anchoredAssets.length; i++) { // some push logic etc }
```
(`forEach` is a newer loop method, but essentially the same thing of using a mutative pattern)

We can do with immutable methods:

```ts
const formattedAnchoredAssets = anchoredAssets
  .filter(asset => asset.type === assetType)
  .map(asset => ({ ...asset, subBalance: new BigNumber(0) }));
// etc
```

Reply:
This isn't new, prob just the file name changed. I can def revisit/refactor today.

Reply:
For the record, I did change this where you commented before to `Array.map`, this is not the same code here. Different file.

Reply:
Going to relocate all this (with the hook using it) to the same balances file. There isn't a reason for it to be in the store.